crates/noirc_evaluator/src/ssa/ssa_gen/mod.rs

```diff
@@ -241,9 +255,27 @@ impl<'a> FunctionContext<'a> {
     }
 
     fn codegen_index(&mut self, index: &ast::Index) -> Values {
-        let array = self.codegen_non_tuple_expression(&index.collection);
+        let array_or_slice = self.codegen_expression(&index.collection).into_value_list(self);
         let index_value = self.codegen_non_tuple_expression(&index.index);
-        self.codegen_array_index(array, index_value, &index.element_type, index.location)
+        if array_or_slice.len() > 1 {
```

Comment:
Here's a good spot to add the "2 values => slice" comment

Reply:
done